server/src/main/java/io/whitefox/core/services/DeltaSharedTable.java

```diff
@@ -2,57 +2,95 @@
   }
 
   public static DeltaSharedTable of(Table table, TableSchemaConverter tableSchemaConverter) {
     var configuration = new Configuration();
-    var dataPath = Paths.get(table.location());
-
-    var dt = DeltaLog.forTable(configuration, dataPath.toString());
-    var snap = dt.update();
-    if (snap.getVersion() == -1) {
+    var dataPath = table.location();
+    var dt = DeltaLog.forTable(configuration, dataPath);
+    if (!dt.tableExists()) {
       throw new IllegalArgumentException(
           String.format("Cannot find a delta table at %s", dataPath));
     }
-    return new DeltaSharedTable(dt, tableSchemaConverter);
+    return new DeltaSharedTable(dt, tableSchemaConverter, table);
   }
 
   public static DeltaSharedTable of(Table table) {
     return of(table, TableSchemaConverter.INSTANCE);
   }
 
   public Optional<Metadata> getMetadata(Optional<String> startingTimestamp) {
-    return getSnapshot(startingTimestamp)
-        .map(snapshot -> new Metadata(
-            snapshot.getMetadata().getId(),
-            Metadata.Format.PARQUET,
-            new TableSchema(tableSchemaConverter.convertDeltaSchemaToWhitefox(
-                snapshot.getMetadata().getSchema())),
-            snapshot.getMetadata().getPartitionColumns()));
+    return getSnapshot(startingTimestamp).map(this::metadataFromSnapshot);
+  }
+
+  private Metadata metadataFromSnapshot(Snapshot snapshot) {
+    return new Metadata(
+        snapshot.getMetadata().getId(),
+        Optional.of(tableDetails.name()),
+        Optional.ofNullable(snapshot.getMetadata().getDescription()),
+        Metadata.Format.PARQUET,
+        new TableSchema(tableSchemaConverter.convertDeltaSchemaToWhitefox(snapshot.getMetadata().getSchema())),
+        snapshot.getMetadata().getPartitionColumns(),
+            snapshot.getMetadata().getConfiguration(),
+        Optional.of(snapshot.getVersion()),
+        Optional.empty(), // size is fine to be empty
+        Optional.empty() // numFiles is ok to be empty here too
+        );
   }
 
   public Optional<Long> getTableVersion(Optional<String> startingTimestamp) {
     return getSnapshot(startingTimestamp).map(Snapshot::getVersion);
   }
 
+  public ReadTableResultToBeSigned queryTable(ReadTableRequest readTableRequest) {
+    Snapshot snapshot;
+    if (readTableRequest instanceof ReadTableRequest.ReadTableCurrentVersion) {
+      snapshot = deltaLog.snapshot();
+    } else if (readTableRequest instanceof ReadTableRequest.ReadTableAsOfTimestamp) {
+      snapshot = deltaLog.getSnapshotForTimestampAsOf(
+          ((ReadTableRequest.ReadTableAsOfTimestamp) readTableRequest).timestamp());
+    } else if (readTableRequest instanceof ReadTableRequest.ReadTableVersion) {
+      snapshot = deltaLog.getSnapshotForVersionAsOf(
+          ((ReadTableRequest.ReadTableVersion) readTableRequest).version());
+    } else {
+      throw new IllegalArgumentException("Unknown ReadTableRequest type: " + readTableRequest);
+    }
```

Comment:
I think it's fine not to manage all the possible parameters for now. but let's write it down, because right now is returing all the files without applying any filtering or checking the pageToken or considering the starting and ending version.